lib/src/realtime_db_pagination.dart

```diff
@@ -260,7 +268,9 @@ class _RealtimeDBPaginationState extends State<RealtimeDBPagination> {
     // To cancel previous live listener when new one is set.
     final tempSub = _liveStreamSub;
 
-    var latestDocQuery = widget.query.limitToFirst(1);
+    var latestDocQuery = widget.descending
+        ? widget.query.limitToLast(1)
+        : widget.query.limitToFirst(1);
     if (_data.isNotEmpty) {
       latestDocQuery = latestDocQuery.endBefore(
         Map<String, dynamic>.from(
```

Comment:
Do you think we also need to update the logic here?

Like `endBefore` should be `startAt` and `_data.first.value` should be `_data.last.value`?

Can you map out how the queries will work to decide if the change is needed.

Reply:
During testing, withuout changing `_data.first.value` I tested by replacing `endBefore` with `startAt` and it did not work.

Reply:
Can you try all the patterns and tell me which one works correctly:

|               | **\_data.first.value** | **\_data.last.value** |
| ------------- | ---------------------- | --------------------- |
| **startAt**   | ❌                     |                       |
| **endBefore** |                        |                       |

Reply:
|  | _data.first.value | _data.last.value |
| ------------- | ------------- | ------------- |
| startAt  | ❌ | ✅  |
| endBefore  | ✅  | ✅  |

Reply:
Looks like we have to use `_data.last.value` and `startAfter` for descending type.